File: src/my_project/easy_problems/from151To200/number_of_lines_to_write_string.py
from typing import List 

class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:

        count: int = 0
        number_lines: int = 1
        print(len(widths))
        for letter in s:
            if count + widths[(ord(letter)%97)] <= 100:
                count += widths[(ord(letter)%97)]
            else: 
                count = widths[(ord(letter)%97)]
                number_lines += 1

        result = [ number_lines, count] 


        return result 


solution = Solution()

print(solution.numberOfLines(widths = [10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10], s = "abcdefghijklmnopqrstuvwxyz"))

print(solution.numberOfLines( widths = [4,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10], s = "bbbcccdddaaa"))

File: my_project_typescript/easy_problems_typescript/from51To100/numberOfLinesToWriteString.ts
function numberOfLines(widths: number[], s: string): number[] {

    let count: number = 0
    let numberLines: number = 1 
    for (let letter of s){
        if (count + widths[letter.charCodeAt(0)%97] <= 100){
            count += widths[letter.charCodeAt(0)%97]
        }
        else {
            count = widths[letter.charCodeAt(0)%97]
            numberLines += 1
        }
    }

    let result: Array<number> = [numberLines, count]
    return result 

};


console.log(numberOfLines([10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10], "abcdefghijklmnopqrstuvwxyz"))

console.log(numberOfLines( [4,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10],  "bbbcccdddaaa"))
